support/cas-server-support-aws-cognito-authentication/src/main/java/org/apereo/cas/authentication/AmazonCognitoAuthenticationAuthenticationHandler.java

```diff
@@ -91,7 +94,14 @@ protected AuthenticationHandlerExecutionResult authenticateUsernamePasswordInter
             attributes.put("userModifiedDate", CollectionUtils.wrap(userResult.userLastModifiedDate().toEpochMilli()));
 
             val userAttributes = userResult.userAttributes();
-            userAttributes.forEach(attr -> attributes.put(attr.name(), CollectionUtils.wrap(attr.value())));
+            userAttributes.forEach(attr -> {
+                if (mappedAttributes != null && !mappedAttributes.isEmpty() && mappedAttributes.containsKey(attr.name())) {
```

Comment:
Can mapped values ever be null?

Reply:
It can not be null. I will remove the != null check.